src/lib/y2storage/device.rb

```diff
@@ -0,0 +1,98 @@
+require "y2storage/storage_class_wrapper"
+
+module Y2Storage
+  # An abstract base class of storage devices and a vertex in the Devicegraph.
+  #
+  # The Device class does not have a device name since some device types do
+  # not have a intrinsic device name, e.g. btrfs.
+  #
+  # This is a wrapper for Storage::Device
+  class Device
+    include StorageClassWrapper
+    wrap_class Storage::Device,
+      downcast_to: ["BlkDevice", "Mountable", "PartitionTables::Base"]
+
+    storage_forward :==
+    storage_forward :!=
+    storage_forward :sid
+    storage_forward :ancestors, as: "Device"
+    storage_forward :exists_in_devicegraph?
+    storage_forward :exists_in_probed?
+    storage_forward :exists_in_staging?
+    storage_forward :displayname
+    storage_forward :detect_resize_info
+    storage_forward :remove_descendants
+    storage_forward :userdata
+    storage_forward :userdata=
+
+    # Checks where the device is a disk
+    #
+    # @return [Boolean] true if this is a Y2Storage::Disk
+    def disk?
```

Comment:
I am not sure if this methods are worth it. What is its advantage? it looks like common ruby methods to get child class.